internal/backends/postgresql/metadata/registry.go

```diff
@@ -701,6 +737,214 @@ func (r *Registry) CollectionRename(ctx context.Context, dbName, oldCollectionNa
 	return true, nil
 }
 
+// IndexesCreate creates indexes in the collection.
+//
+// Existing indexes with given names are ignored.
+func (r *Registry) IndexesCreate(ctx context.Context, dbName, collectionName string, indexes []IndexInfo) error {
+	defer observability.FuncCall(ctx)()
+
+	p, err := r.getPool(ctx)
+	if err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	r.rw.Lock()
+	defer r.rw.Unlock()
+
+	return r.indexesCreate(ctx, p, dbName, collectionName, indexes)
+}
+
+// indexesCreate creates indexes in the collection.
+//
+// Existing indexes with given names are ignored.
+//
+// It does not hold the lock.
+func (r *Registry) indexesCreate(ctx context.Context, p *pgxpool.Pool, dbName, collectionName string, indexes []IndexInfo) error {
+	defer observability.FuncCall(ctx)()
+
+	_, err := r.collectionCreate(ctx, p, dbName, collectionName)
+	if err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	db := r.colls[dbName]
+	if db == nil {
+		panic("database does not exist")
+	}
+
+	c := r.collectionGet(dbName, collectionName)
+	if c == nil {
+		panic("collection does not exist")
+	}
+
+	created := make([]string, 0, len(indexes))
+
+	for _, index := range indexes {
+		if slices.ContainsFunc(c.Settings.Indexes, func(i IndexInfo) bool { return index.Name == i.Name }) {
+			continue
+		}
+
+		// indexes must be unique across the whole database, so we add a uuid to the index name
+		uuidPart := must.NotFail(uuid.NewRandom()).String()
+		tableNamePart := c.TableName
+		tableNamePartMax := maxIndexNameLength - len(uuidPart) - 5 // 5 is for _ and _idx
```

Comment:
Let's try to not introduce a constant when we can get away without it :hugs: 

```suggestion
		tableNamePart := fmt.Sprintf("_%s",c.TableName)
		indexSuffixPart := "_idx"
		tableNamePartMax := maxIndexNameLength - len(tableNamePart) - len(uuidPart) - len(indexSuffixPart) 
```

Then in `index.TableIndexName` we can just assign concat of string.

```
index.TableIndexName = fmt.Sprintf("%s%s%s", tableNamePart, uuidPart, indexSuffixPart)
```